src/Bridge/Graphql/Resolver/ItemResolverFactory.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Graphql\Resolver;
+
+use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
+use GraphQL\Type\Definition\ResolveInfo;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Creates a function retrieving an item to resolve a GraphQL query.
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class ItemResolverFactory implements ItemResolverFactoryInterface
+{
+    private $itemDataProvider;
+    private $normalizer;
+
+    public function __construct(ItemDataProviderInterface $itemDataProvider, NormalizerInterface $normalizer)
+    {
+        $this->itemDataProvider = $itemDataProvider;
+        $this->normalizer = $normalizer;
+    }
+
+    public function createItemResolver(string $resourceClass, string $operationName): callable
+    {
+        return function ($root, $args, $context, ResolveInfo $resolveInfo) use ($resourceClass, $operationName) {
+            $item = $this->itemDataProvider->getItem($resourceClass, $args['id'], $operationName);
+            if (null === $item) {
+                return null;
+            }
+            $selectedFields = $resolveInfo->getFieldSelection();
+
+            return $this->normalizer->normalize($item, null, ['graphql_selected_fields' => $selectedFields]);
```

Comment:
One line:
```php
return ($item = $itemDataProvider->getItem($resourceClass, $args['id'], $operationName))
    ? $normalizer->normalize($item, null, ['graphql_selected_fields' => $resolveInfo->getFieldSelection()])
    : null;
```

Reply:
Sometimes, I prefer to be more verbose to avoid potential bugs and to modify more easily the code. But if it is preferred in API Platform, no problem.